frontend/src/features/modalSlice.ts

```diff
@@ -0,0 +1,35 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { RootState } from '../app/store';
+
+export interface ModalState {
+  modalType: string | null;
+  modalProps: Object;
+}
+
+const initialState: ModalState = {
+  modalType: null,
+  modalProps: {},
+};
+
+export const modalSlice = createSlice({
```

Comment:
Also not sure if using the redux store to handle opening/closing the modal is best, unless we need other components to know that the modal is open/closed. 

I've been told that if its only that component that needs to know then just use react state. Which would be stored in the parent the modal, the employees page in this case.

Reply:
So this method was actually suggested by my REACT mentor from my training. 
It also feels like the employee component is handling a lot of things rn, which isn't ideal. 

This method minimises work in the future too. I also imagine its quite likely we will get modals in someway coming up in other compoents, such as redifinng or deleting someones shifts?